demo-cloud-migration.md

```diff
@@ -0,0 +1,199 @@
+---
+title: Cloud Migration
+summary: Use a local cluster to simulate migrating from one cloud platform to another.
+toc: false
+---
+
+CockroachDB's flexible [replication controls](configure-replication-zones.html) make it trivially easy to run a single CockroachDB cluster across cloud platforms or to migrate a cluster from one cloud to another without any service interruption. This page walks you through a local simulation of the process.
+
+<div id="toc"></div>
+
+## Before You Begin
+
+Make sure you have already [installed CockroachDB](install-cockroachdb.html).
+
+## Step 1. Start a 3-node cluster on "cloud 1"
+
+If you've already [started a local cluster](start-a-local-cluster.html), these commands should be familiar to you. The new flag to note is [`--locality`](configure-replication-zones.html#descriptive-attributes-assigned-to-nodes), which accepts key-value pairs that describe the locality of a node. In this case, you're using the flag to specify that the first 3 nodes are running on "cloud platform 1".
+
+~~~ shell
+# In a new terminal, start node 1 on cloud 1:
+$ cockroach start --insecure \
+--locality=cloud=platform1 \
+--store=cloud1node1 \
+--host=localhost \
+--cache=100MB
+
+# In a new terminal, start node 2 on cloud 1:
+$ cockroach start --insecure \
+--locality=cloud=platform1 \
+--store=cloud1node2 \
+--host=localhost \
+--port=25258 \
+--http-port=8081 \
+--join=localhost:26257 \
+--cache=100MB
+
+# In a new terminal, start node 3 on cloud 1:
+$ cockroach start --insecure \
+--locality=cloud=platform1 \
+--store=cloud1node3 \
+--host=localhost \
+--port=25259 \
+--http-port=8082 \
+--join=localhost:26257 \
+--cache=100MB
+~~~
+
+## Step 2. Set up HAProxy load balancing
+
+Each CockroachDB node is an equally suitable SQL gateway to your cluster, but to ensure client performance and reliability, it's important to use TCP load balancing. [HAProxy](http://www.haproxy.org/) is one of the most popular open-source TCP load balancers, and CockroachDB includes a built-in command for generating a configuration file that is preset to work with your running cluster, so you'll use that tool here.
```

Comment:
How about: "..., but to ensure an even balancing of client requests across available nodes, we can use a TCP load balancer."